newIDE/app/src/Profile/ProfileDetails.js

```diff
@@ -4,67 +4,121 @@ import { Trans, t } from '@lingui/macro';
 import * as React from 'react';
 import Avatar from '@material-ui/core/Avatar';
 import { Column, Line, Spacer } from '../UI/Grid';
-import { type Profile } from '../Utils/GDevelopServices/Authentication';
 import PlaceholderLoader from '../UI/PlaceholderLoader';
 import { getGravatarUrl } from '../UI/GravatarUrl';
 import Text from '../UI/Text';
 import RaisedButton from '../UI/RaisedButton';
 import TextField from '../UI/TextField';
 import { I18n } from '@lingui/react';
+import FlatButton from '../UI/FlatButton';
+import { ColumnStackLayout } from '../UI/Layout';
+import AlertMessage from '../UI/AlertMessage';
+import { type AuthenticatedUser } from './AuthenticatedUserContext';
 
 type Props = {|
-  profile: ?Profile,
   onEditProfile: Function,
+  authenticatedUser: AuthenticatedUser,
 |};
 
-export default ({ profile, onEditProfile }: Props) => {
-  return profile ? (
+export default ({ onEditProfile, authenticatedUser }: Props) => {
+  const profile = authenticatedUser.profile;
+  const firebaseUser = authenticatedUser.firebaseUser;
+  const [emailSent, setEmailSent] = React.useState<boolean>(false);
+  const sendEmail = React.useCallback(
+    () => {
+      authenticatedUser.onSendEmailVerification();
+      setEmailSent(true);
+      setTimeout(() => setEmailSent(false), 3000);
```

Comment:
We're "leaking" this timeout, in the sense that if I close the dialog it will fire after 3 seconds. Not a huge problem (react will probably still complain we try to modify the state of a closed component).
Would still maybe be a good idea to use `useIsMounted`, and in the setTimeout callback, check `if (!isMounted.current) return;`